sdk/servicebus/azure-servicebus/azure/servicebus/_base_handler.py

```diff
@@ -236,9 +235,6 @@ def _do_retryable_operation(self, operation, timeout=None, **kwargs):
 
     def _mgmt_request_response(self, mgmt_operation, message, callback, keep_alive_associated_link=True, **kwargs):
         self._open()
-        if not self._running:
```

Comment:
We probably need to make a principled decision at some point as to whether we allow operating on a non-opened client or not.  May I ask the impetus for yanking this now?

Reply:
@KieranBrantnerMagee `self._open()` would either succeed to open the underlying connection or raise error out.
So actually `if not self._running: raise InvalidHandlerState("Client connection is closed.")` would never be executed.

As you mentioned, we'd better sync up with languages or probably with service team whether we want sender/receiver to be reusable, if not then what's the reason behind that.

Reply:
Thanks Adam. I was removing InvalidHandlerState, which is not meaningful to users. Then I found this place used it but it was actually unreachable code.